trieste/bayesian_optimizer.py

```diff
@@ -43,6 +44,9 @@
 SP = TypeVar("SP", bound=SearchSpace)
```

Comment:
more descriptive name, e.g. SearchSpaceType?

Reply:
It's standard in Python to use initials for type variables: e.g. see https://www.python.org/dev/peps/pep-0008/#type-variable-names. Happy to discuss whether it makes sense to break this convention, but probably not as part of this PR?